File: frontend/lib/models/me.dart
/// Espelha MeResponse em backend/app/routers/schemas.py — GET /me.
class Me {
  final String id;
  final String email;
  final int totalXp;
  final int currentStreak;
  final int longestStreak;

  const Me({
    required this.id,
    required this.email,
    required this.totalXp,
    required this.currentStreak,
    required this.longestStreak,
  });

  factory Me.fromJson(Map<String, dynamic> json) {
    return Me(
      id: json['id'] as String,
      email: json['email'] as String,
      totalXp: json['total_xp'] as int,
      currentStreak: json['current_streak'] as int,
      longestStreak: json['longest_streak'] as int,
    );
  }
}

File: backend/app/routers/me.py
"""`GET /me` — perfil do usuário logado (RF-01)."""

from fastapi import APIRouter, Depends

from app.models.entities import AppUser
from app.routers.deps import get_current_app_user
from app.routers.schemas import MeResponse

router = APIRouter(tags=["me"])


@router.get("/me", response_model=MeResponse)
def get_me(user: AppUser = Depends(get_current_app_user)) -> MeResponse:
    return MeResponse(
        id=user.id,
        email=user.email,
        total_xp=user.total_xp,
        current_streak=user.current_streak,
        longest_streak=user.longest_streak,
    )
